bdb/rep.c

```diff
@@ -2116,11 +2113,12 @@ void bdb_get_notcoherent_list(bdb_state_type *bdb_state,
     *num_notcoherent = 0;
     for (i = 0; i < count; i++) {
         if (is_incoherent(bdb_state, hostlist[i])) {
-            {
-                if ((*num_notcoherent) < max_nodes)
-                    nodes_list[*num_notcoherent] = hostlist[i];
-                (*num_notcoherent)++;
-            }
+            if ((*num_notcoherent) < max_nodes)
+                nodes_list[*num_notcoherent] = hostlist[i];
+            (*num_notcoherent)++;
+
+            if (bdb_state->repinfo->skipsinceepoch == 0)
+                abort();
         }
```

Comment:
There are no circumstances where skipsinceepoch can legitimately be zero here?

Reply:
I guess i was trying to assert that skipsinceepoch is non zero. This check is not needed, i'm removing it -- we don't gain anything from it. skipsinceepoch is set every time a node is not coherent.